ansifilter: fix compiler incompatibility on OSX 10.8 and below

tessus opened the pull request:
Older compilers do not have the C++11 standard included.
ansilfilter uses unique_ptr, which is only available in C++11.
This patch removes the C++11 flag from the compiler options and
replaces the unique_ptr with an auto_ptr type.

tobypeterson commented:
Looks quite reasonable. Thanks!

tessus commented:
@tobypeterson thanks, it was close. 

still something off though. argh. I really need access to those build machines.... !!!!! it's almost impossible to fix those things by trial and error.

Do you have a VM image (Parallels, Virtual Box, ...) for osx 10.7 or 10.8 with macports installed, which I can download? In that case I can play around until it works, before I submit another fix.

tobypeterson commented:
Not publicly accessible VMs. Sorry. Seems like explicitly passing -std=c++03 and fixing the issues would suffice. Unless that's already what you did...

tobypeterson commented:
Actually that doesn't do anything even if I leave in unique_ptr. Guess the compiler doesn't care. This looks like a pretty straightforward project, will try on a VM when I can...

tessus commented:
I just removed the c++11 flag, which is basically the same as setting c++03. then I replaced the pointer type, since unique_ptr is only available in c++11.

but I ran into another issue:

```
rtfgenerator.cpp:181:12: error: use of undeclared identifier 'to_string'
      m += to_string(utf16Char);
```
and
```
rtfgenerator.cpp: In member function ‘virtual std::string ansifilter::RtfGenerator::maskCharacter(unsigned char)’:
rtfgenerator.cpp:181: error: ‘to_string’ was not declared in this scope
rtfgenerator.cpp: In member function ‘std::string ansifilter::RtfGenerator::unicodeFromHTML(const std::string&)’:
rtfgenerator.cpp:314: error: ‘to_string’ was not declared in this scope
```

It seems that to_string is either another c++11 extension or something else is amiss.

Without a machine to actually test a build, I don't know what to do.

tessus commented:
I don't really want to bother you with this stuff. Maybe I find somebody at work who has still an ancient OSX on a spare machine I can use....

kencu commented:
The usual fix for c++11 needs is to add the cxx11 portgroup, which takes care of ensuring it builds on a system that can handle cxx11. cxx11 compilers are available all the way back to Tiger (gcc versions), and that portgroup (there are two, 1.0, and a newer 1.1) should fix this all up for you automatically.

tessus commented:
@kencu thanks for the info. I was actually looking for something like this in the documentation, but I couldn't find anything. Where in the docs did you find this?
What happens on a client system? Will a c++11 compatible compiler be compiled and installed? 

Also, what is the difference between 1.0 and 1.1? When would I choose one over the other?

So basically I can remove the patches and add the portgroup at the top of the port file, right?

tessus commented:
@kencu @tobypeterson I made the changes. does this look right? see #428

kencu commented:
The cxx11 1.1 portgroup invokes certain tricks for clang-3.9 and clang-4.0 that make it build c++11 software on the buildbots with libstdc++. This works well, and  I think the party line these days would be to use the cxx11 1.1 portgroup.

tessus commented:
thanks for the info. how does my change #428 look like? do you think that this will make it work on machines < osx 10.8?

kencu commented:
If it is "UsingTheRightCompiler" it should "just work". 

Some Makefiles need to have cc=${configure.cc} or similar injected to make it all work out, as Toby points out elsewhere. I guess the buildbots will tell us shortly!

tobypeterson commented:
Yeah, looks like it needs:

`build.args-append CC=${configure.cxx}`

At least, that makes it use /usr/bin/clang++ on my 10.12.x machine.

tobypeterson commented:
Probably needs configure.cxxflags as well, since those seem to be set by the cxx11 portgroup, presumably they are helpful.

tessus commented:
great, I'll add them in a sec

tessus commented:
just checked the makefile of ansifilter. it has no CXXFLAGS var, but I could append it to CFLAGS.
